src/adapt.jl

```diff
@@ -32,7 +35,8 @@ function do_quadgk(f::F, s::NTuple{N,T}, n, atol, rtol, maxevals, nrm) where {T,
         return (I, E) # fast return when no subdivisions required
     end
 
-    return adapt(f, heapify!(collect(segs), Reverse), I, E, numevals, x,w,gw,n, atol_, rtol_, maxevals, nrm)
+    heapify!(segbuf, Reverse)
```

Comment:
Maybe instead, you can keep the above code the same (i.e., still use tuples), and then change this line to:
```
segheap = isnothing(segbuf) ? collect(segs) : (resize!(segbuf, N-1) .= segs)
heapify!(segheap, Reverse)
```
and then pass `segheap` to adapt

Reply:
Thanks for the feedback. Seems like a good idea and nice way of implementing it. Done!